tools/examples/src/main.ts

```diff
@@ -0,0 +1,115 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { lstatSync, readdirSync } from 'fs';
+import { join } from 'path';
+import { bold, green } from 'chalk';
+import {
+  getExamplePackageMetadata,
+  ExamplePackageMetadata,
+  ExampleMetadata,
+} from './metadata';
+import { generateExamplesLibBarrelFile } from './generate-examples-lib-barrel';
+import { generateDemosAppExamplesModule } from './generate-examples-module';
+import { generateDemosAppRoutingModule } from './generate-routing-module';
+import { generateDemosAppNavItems } from './generate-nav-items';
+
+export const EXAMPLES_ROOT = join(
+  __dirname,
+  '../../../',
+  'libs',
+  'examples',
+  'src',
+);
+export const DEMOS_APP_ROOT = join(
+  __dirname,
+  '../../../',
+  'apps',
+  'demos',
+  'src',
+);
+
+/** Collect all files containing examples in the demos app. */
+async function getExamplesInPackages(): Promise<ExamplePackageMetadata[]> {
+  return (await Promise.all(
+    readdirSync(EXAMPLES_ROOT)
+      .map(name => join(EXAMPLES_ROOT, name))
+      .filter(dir => lstatSync(dir).isDirectory())
+      .map(dir => getExamplePackageMetadata(dir)),
+  )).filter(Boolean) as ExamplePackageMetadata[];
+}
+
+async function main(): Promise<void> {
+  console.log();
+  console.log(bold(`Generating Barista Examples`));
+  console.log(`Collecting example and module files and extracting metadata`);
+  const packageMetas = await getExamplesInPackages();
+  const examples = packageMetas.reduce<ExampleMetadata[]>(
+    (aggregatedExamples, packageMeta) => [
+      ...aggregatedExamples,
+      ...packageMeta.examples,
+    ],
+    [],
+  );
+  console.log(
+    green(
+      `  ✓   ${packageMetas.length} packages with ${examples.length} examples found`,
+    ),
+  );
+
+  console.log(`Generating lib barrel file`);
+  const rootBarrelFile = await generateExamplesLibBarrelFile(
+    packageMetas,
+    EXAMPLES_ROOT,
+  );
+  console.log(green(`  ✓   Created "${rootBarrelFile}"`));
+
+  console.log(`Generating examples module for demos app`);
+  const demosModuleFile = await generateDemosAppExamplesModule(
+    packageMetas,
+    DEMOS_APP_ROOT,
+  );
+  console.log(green(`  ✓   Created "${demosModuleFile}"`));
+
+  console.log(`Generating demos app routes & routing module`);
+  const routingModule = await generateDemosAppRoutingModule(
+    examples,
+    DEMOS_APP_ROOT,
+  );
+  console.log(green(`  ✓   Created "${routingModule}"`));
+
+  console.log(`Generating nav-items for the demos app menu`);
+  const navItemsFile = await generateDemosAppNavItems(
+    packageMetas,
+    DEMOS_APP_ROOT,
+  );
+  console.log(green(`  ✓   Created "${navItemsFile}"`));
+
+  console.log(
+    bold(
+      green(
+        `✓ All content for the demos & barista app has been successfully generated`,
+      ),
+    ),
+  );
+  console.log();
+}
+
+main()
+  .then(() => {})
```

Comment:
Usually you return a number (status code of your application in the main 

please add a process.exit(0)
```suggestion
  .then((code: 0 | 1) => { process.exit(code); })
```